pandas/core/indexes/interval.py

```diff
@@ -1140,7 +1140,7 @@ def overlaps(self, other):
 
     @Appender(_index_shared_docs['intersection'])
     @SetopCheck(op_name='intersection')
-    def intersection(self, other, sort=False):
+    def intersection(self, other, sort: bool = False):
```

Comment:
can you add typing on other? (and do this for all the intersection calls)?

Reply:
I did try something as follows in HEAD~1:
```
    def intersection(self,
                     other: IntervalIndex,
                     sort: bool = False
                     ) -> IntervalIndex:
```
But this will run into the NameError, i.e.,
`NameError: name 'IntervalIndex' is not defined`

Reply:
use the string Intervalndex and this will work